### ISSUE TITLE
ClickHouse dialect does not support final modifier

### ISSUE THREAD

**Issue by jbradt** [sqlfluff/sqlfluff#3507]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

ClickHouse uses a `final` modifier on table names in some queries. This modifier appears after the table name in a simple SELECT query. SQLFluff misinterprets this keyword as a table alias.

### Expected Behaviour

SQLFluff should understand that `final` is not an alias in this context.

### Observed Behaviour

Given the query

```sql
select a
from my_table final
where a > 0;
```

SQLFluff produces the warnings:
```
❯ sqlfluff lint example.sql --dialect clickhouse
== [example.sql] FAIL
L:   2 | P:  15 | L011 | Implicit/explicit aliasing of table.
L:   2 | P:  15 | L025 | Alias 'final' is never used in SELECT statement.
L:   2 | P:  15 | L029 | Keywords should not be used as identifiers.
L:   2 | P:  15 | L031 | Avoid aliases in from clauses and join conditions.
All Finished 📜 🎉!
```

Notice how it seems to think `final` is an alias, leading to surprising warnings.

### How to reproduce

```sql
-- example.sql
select a
from my_table final
where a > 0;
```

Command:
```
sqlfluff lint example.sql --dialect clickhouse
```

### Dialect

ClickHouse

### Version

```
❯ sqlfluff --version
sqlfluff, version 1.0.0

❯ python --version
Python 3.7.13
```

### Configuration

Just the stock config, no other files.

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by tunetheweb** [comment]
Should be a simple matter of overriding `class FromExpressionElementSegment` in the dialect_clickhouse.py file and adding `Ref.keyword("Final", optional=True),` after `OptionallyBracketed(Ref("TableExpressionSegment")),` and then adding a test case as detailed in [Contributing Dialect Changes wiki](https://github.com/sqlfluff/sqlfluff/wiki/Contributing-Dialect-Changes) if anyone fancies a stab at this.

**Comment by ThomAub** [comment]
With such a detailed issue and hint i took a look and gave it a go in a corresponding PR.

**Comment by WittierDinosaur** [comment]
Closing as this issue isn't present in the current version of SQLFluff
